src/main/java/org/cirdles/topsoil/data/NumberField.java

```diff
@@ -43,7 +44,13 @@ public String toString(Number number) {
 
             @Override
             public Number fromString(String string) {
-                return Double.valueOf(string);
+                
+               //Check if the string parameter is a number
+                if (string.matches("\\d+(\\.\\d+)?")) {
```

Comment:
This actually needs to be fixed as well. Some testing for this might be good too. Isn't `-45` a number?